File: app/src/main/java/dev/blaauwendraad/recipe_book/service/model/Recipe.java
package dev.blaauwendraad.recipe_book.service.model;

import jakarta.annotation.Nullable;
import java.util.List;

public record Recipe(
        Long id,
        String title,
        @Nullable String description,
        Integer numServings,
        PreparationTime preparationTime,
        @Nullable Author author,
        List<Ingredient> ingredients,
        List<PreparationStep> preparationSteps) {}

File: ui/src/models/domain/Recipe.ts
import type Ingredient from "./Ingredient.ts";
import type PreparationStep from "./PreparationStep.ts";

export default interface Recipe {
    id: number;
    title: string;
    description: string;
    author: {
        userId: number;
        username: string;
    };
    ingredients: Ingredient[];
    preparationSteps: PreparationStep[];
}
